src/graphql/schema/collection/createConnectionFieldFromPaginator.ts

```diff
@@ -0,0 +1,90 @@
+import { GraphQLFieldConfig, GraphQLInputObjectType, GraphQLInputFieldConfig } from 'graphql'
+import { Collection, Condition, conditionHelpers, NullableType, ObjectType, Paginator, Relation } from '../../../interface'
+import { formatName, buildObject, memoize2 } from '../../utils'
+import BuildToken from '../BuildToken'
+import getGqlType from '../getGqlType'
+import createConnectionGqlField from '../connection/createConnectionGqlField'
+import transformGqlInputValue from '../transformGqlInputValue'
+
+const getConditionGqlTypeAndConditionFieldEntries = memoize2((buildToken: BuildToken, type: ObjectType) => {
+  // Creates the field entries for our paginator condition type.
+  const gqlConditionFieldEntries =
+    Array.from(type.fields).map<[string, GraphQLInputFieldConfig<mixed> & { internalName: string }]>(([fieldName, field]) =>
+      [formatName.field(fieldName), {
+        description: `Checks for equality with the object’s \`${formatName.field(fieldName)}\` field.`,
+        // Get the type for this field, but always make sure that it is
+        // nullable. We don’t want to require conditions.
+        type: getGqlType(buildToken, new NullableType(field.type), true),
+        // We include this internal name so that we can resolve the arguments
+        // back into actual values.
+        internalName: fieldName,
+      }],
+    )
+
+  // Creates our GraphQL condition type.
+  const gqlConditionType = new GraphQLInputObjectType({
+    name: formatName.type(`${type.name}-condition`),
+    description: `A condition to be used against \`${formatName.type(type.name)}\` object types. All fields are tested for equality and combined with a logical ‘and.’`,
+    fields: buildObject<GraphQLInputFieldConfig<mixed>>(gqlConditionFieldEntries),
+  })
+
+  return {gqlConditionType, gqlConditionFieldEntries}
```

Comment:
From here perhaps we should return (the names can be aliased in consuming functions):

```ts
{
  gqlType: GraphQLInputObjectType,
  fromGqlInput: (input: any) => Condition,
}
```

This way we don’t have to directly return the field entries (as the field entries will only be used in `fromGqlInput`). The implementation of the `fromGqlInput` function will mostly be the current implementation you have for `getPaginatorInput` (minus the relation bit).